### PR TITLE
Add metric observer and remove setCallback from Metric base class

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/metrics/Meter.java [hunk 1]
```diff
@@ -191,6 +191,32 @@
    */
   MeasureLong.Builder measureLongBuilder(String name);
 
+  /**
+   * Returns a new builder for a {@link MeasureDouble}.
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/metrics/Meter.java]:
```suggestion
   * Returns a new builder for a {@link ObserverDouble}.
```

Same for all others.

File: api/src/main/java/io/opentelemetry/metrics/Meter.java [hunk 2]
```diff
@@ -191,6 +191,32 @@
...
+   *
+   * @param name Name of measure, as a {@code String}. Should be a ASCII string with a length no
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/metrics/Meter.java]:
```suggestion
   * @param name Name of observer, as a {@code String}. Should be a ASCII string with a length no
```

File: api/src/main/java/io/opentelemetry/metrics/Meter.java [hunk 3]
```diff
@@ -191,6 +191,32 @@
...
+   * @return a new builder for a {@code MeasureDouble}.
+   * @throws NullPointerException if {@code name} is null.
+   * @throws IllegalArgumentException if different metric with the same name already registered.
+   * @throws IllegalArgumentException if the {@code name} does not match the requirements.
+   * @since 0.1.0
+   */
+  ObserverDouble.Builder observerDoubleBuilder(String name);
+
+  /**
+   * Returns a new builder for a {@link MeasureLong}.
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/metrics/Meter.java]: Ditto

File: api/src/main/java/io/opentelemetry/metrics/Observer.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  /**
+   * A {@code Handle} for a {@code ObserverDouble}.
+   *
+   * @since 0.1.0
+   */
+  @ThreadSafe
+  interface Handle {}
+
+  /**
+   * A {@code Handle} for a {@code MeasureLong}.
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/metrics/Observer.java]:
```suggestion
   * A {@code Callback} for a {@code Observer}.
```

File: api/src/main/java/io/opentelemetry/metrics/Observer.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * Base interface for all the Observer metrics.
+ *
+ * @param <R> the Handle.
+ * @since 0.1.0
+ */
+public interface Observer<R> extends Metric<Observer.Handle> {
+  /**
+   * A {@code Handle} for a {@code ObserverDouble}.
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/metrics/Observer.java]:
```suggestion
   * A {@code Handle} for a {@code Observer}.
```

File: api/src/main/java/io/opentelemetry/metrics/Meter.java (hunk 3 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/Meter.java]: Done.

File: api/src/main/java/io/opentelemetry/metrics/Meter.java [hunk 6]
```diff
@@ -191,6 +191,32 @@
    */
   MeasureLong.Builder measureLongBuilder(String name);
 
+  /**
+   * Returns a new builder for a {@link ObserverDouble}.
+   *
+   * @param name Name of observer, as a {@code String}. Should be a ASCII string with a length no
+   *     greater than 255 characters.
+   * @return a new builder for a {@code MeasureDouble}.
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/metrics/Meter.java]:
```suggestion
   * @return a new builder for a {@code ObserverDouble}.
```

File: api/src/main/java/io/opentelemetry/metrics/Meter.java [hunk 7]
```diff
@@ -191,6 +191,32 @@
...
+   * @since 0.1.0
+   */
+  ObserverDouble.Builder observerDoubleBuilder(String name);
+
+  /**
+   * Returns a new builder for a {@link ObserverLong}.
+   *
+   * @param name Name of observer, as a {@code String}. Should be a ASCII string with a length no
+   *     greater than 255 characters.
+   * @return a new builder for a {@code MeasureLong}.
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/metrics/Meter.java]:
```suggestion
   * @return a new builder for a {@code ObserverLong}.
```

**Pull request merged**